digdag-standards/src/main/java/io/digdag/standards/operator/td/TdDdlOperatorFactory.java

```diff
@@ -110,8 +116,35 @@ public TaskResult runTask()
                     op.withDatabase(t.getDatabase().or(op.getDatabase())).ensureTableCreated(t.getTable());
                 });
             }
+            for (RenameTableConfig r : renameTableList) {
+                TableParam from = r.getFromTable();
+                String to = r.getToTable();
+                if (to.contains(".")) {
+                    // renaming a table across databases is not supported by the td's rest api itself
+                    throw new ConfigException("'to' option of rename_tables can't include database name");
+                }
+                operations.add(op -> {
+                    logger.info("Renaming TD table {}.{} -> {}", op.getDatabase(), from, to);
+                    op.withDatabase(from.getDatabase().or(op.getDatabase())).ensureExistentTableRenamed(from.getTable(), to);
+                });
+            }
 
             try (TDOperator op = TDOperator.fromConfig(env, params, context.getSecrets().getSecrets("td"))) {
+                // make sure that all "from" tables exist so that ignoring 404 Not Found in
+                // op.ensureExistentTableRenamed is valid.
+                boolean renameFromChecked = state.params().get("rename_from_checked", boolean.class, false);
```

Comment:
Wrap in `PollingRetryExecutor.runOnce` to avoid repeating this check for every invocation. Otherwise this check might fail due to partial success and polling retry below.